practicecards.js

```diff
@@ -0,0 +1,63 @@
+import React from 'react'
+
+export default class PracticeCards extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      currentCard: 0,
+      showAnswer: false
+    }
+    this.toggleAnswer = this.toggleAnswer.bind(this)
+  }
+
+  toggleAnswer() {
+    this.setState({
+      showAnswer: !this.state.showAnswer
+    })
+  }
+
+  changeCard(direction) {
+    let indexToUpdate = this.state.currentCard
+    if (direction === 'left') {
+      indexToUpdate--
+    }
+    if (direction === 'right') {
+      indexToUpdate++
+    }
+    this.setState({
+      currentCard: indexToUpdate,
+      showAnswer: false
+    })
+  }
+
+  mod(x, n) {
```

Comment:
What is this?

Reply:
It seems to do more than `mod`. If you are getting the next card index, you should name this method something like `getNextCardIndex`.

Reply:
When Javascript takes the mod of a negative number, it returns a negative integer instead of returning the positive mod.  (-1) % 4 in js gives you -1, not 3 as would be expected.  This is a workaround I found that allows for the wrap-around effect in a carousel, so that the following is true:
indexes: [-2] [-1] [0] [1] [2] 
ind %4:  [ 2 ] [ 3 ] [0] [1] [2]

Reply:
I will rename it to make its intention more clear